### PR TITLE
Core changed requested by Language Server for Alloy

### PR DESCRIPTION
**Pull request by pkriens**
Arash Sahebolamri submitted PR #162. This provides a Language Server for Alloy. However, this PR made some significant changes in the core.

I've separated the core changes from the Language Server changes and put the server changes in a separate project. 

This PR introduces an Assert class and makes many small changes.

### CODE REVIEW

**Comment by grayswandyr** [comment]
Pinging @nmacedo who's more relevant than me.

**Comment by pkriens** [comment]
@aleksandarmilicevic @nmacedo @grayswandyr can anyone approve this? I don't think the changes are very extensive, nor seem to influence the language, and I need them to integrate the LSP project ... and then the shell.

File: org.alloytools.alloy.core/src/test/java/org/alloytools/alloy/core/AlloyModelsTest.java [hunk 1]
```diff
@@ -48,4 +50,40 @@ public void minimalAlloyDoc() throws Exception {
...
+            //cmd.additionalExactScopes.forEach(sig -> sig.accept(visitor));
+            cmd.formula.accept(visitor);
+//            cmd.scope.forEach(scope -> {
+//                scope.sig.accept(visitor);
+//            });
+
+        });
+        //world.visitExpressions(new VisitQuery<String>() {});
+
+        //Expr expr = world.find(new Pos(null, 12, 3, 15, 3));
```
Comment (aleksandarmilicevic) [inline org.alloytools.alloy.core/src/test/java/org/alloytools/alloy/core/AlloyModelsTest.java:85]: delete all the commented out code in this file

**Review by aleksandarmilicevic — APPROVED** [review]
LGTM

**Pull request merged**